src/btree/bt_cursor.c

```diff
@@ -1094,7 +1093,8 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
      * not be positioned to the correct record in the case of implicit records in the append list.
      * FUTURE: again, it appears that this is no longer true...
      */
-    if (btree->type != BTREE_COL_FIX && __cursor_page_pinned(cbt, false)) {
+
+    if (session == NULL && btree->type != BTREE_COL_FIX && __cursor_page_pinned(cbt, false)) {
```

Comment:
Empty line you don't need.

The `session == NULL` is pretty awful (I was just hacking stuff), :).
Maybe: 
```
index 550ac8de4..c92e46c7b 100644
--- a/src/btree/bt_cursor.c
+++ b/src/btree/bt_cursor.c
@@ -1074,6 +1074,7 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
     /* Save the cursor state. */
     __cursor_state_save(cursor, &state);
 
+#if 0
     /*
      * If remove positioned to an on-page key, the remove doesn't require another search. We don't
      * care about the "overwrite" configuration because regardless of the overwrite setting, any
@@ -1093,7 +1094,6 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
      * not be positioned to the correct record in the case of implicit records in the append list.
      * FUTURE: again, it appears that this is no longer true...
      */
-
     if (session == NULL && btree->type != BTREE_COL_FIX && __cursor_page_pinned(cbt, false)) {
         WT_ERR(__wt_txn_autocommit_check(session));
 
@@ -1108,6 +1108,7 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
             goto done;
         goto err;
     }
+#endif
 
 retry:
     /*
```

If we eventually remove this entirely, we're going to need to generally clean up this function.

Reply:
Yeah, so the reason I went with your hack is because when I use the `#if` and `endif` tags is because the compiler complains that we are not using the `done:` tag when disabling the code path.